### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by igorng**
[ALIEN-2125]  

- process inputs before location and node matchings / substitutions
- updated deployment view to display a section related to the active deployment (inputs and outputs)
- fixed spring cyclic dependency on  loading ResourcePermissionService

### CODE REVIEW

File: alien4cloud-core/src/main/java/alien4cloud/deployment/DeploymentTopologyService.java [hunk 1]
```diff
@@ -294,6 +308,10 @@ private void doUpdateDeploymentTopology(DeploymentTopology deploymentTopology, T
     public void updateDeploymentTopologyInputsAndSave(DeploymentTopology deploymentTopology) {
         deploymentInputService.processInputProperties(deploymentTopology);
         deploymentInputService.processProviderDeploymentProperties(deploymentTopology);
+        /// since inputs must be done before processing substitutions, we do it here
```
Comment (lucboutier) [inline alien4cloud-core/src/main/java/alien4cloud/deployment/DeploymentTopologyService.java]: Should we do that if there is no location matched yet ?

File: alien4cloud-core/src/main/java/alien4cloud/deployment/DeploymentInputService.java [hunk 2]
```diff
@@ -36,6 +36,9 @@
     @Inject
     private OrchestratorDeploymentService orchestratorDeploymentService;
 
+    @Inject
+    private InputsPreProcessorService inputsPreProcessorService;
```
Comment (lucboutier) [inline alien4cloud-core/src/main/java/alien4cloud/deployment/DeploymentInputService.java]: Why ? This is not used anywhere...

File: alien4cloud-core/src/main/java/alien4cloud/deployment/DeploymentTopologyService.java (hunk 1 above)
Comment (igorng) [inline alien4cloud-core/src/main/java/alien4cloud/deployment/DeploymentTopologyService.java]: good point. We could indeed check if a location has already been selected for the deploymentTopology before  injecting inputs and processing substitutions.

File: alien4cloud-core/src/main/java/alien4cloud/deployment/DeploymentInputService.java (hunk 2 above)
Comment (igorng) [inline alien4cloud-core/src/main/java/alien4cloud/deployment/DeploymentInputService.java]: oups.. :)

File: alien4cloud-core/src/main/java/alien4cloud/deployment/InputsPreProcessorService.java [hunk 3]
```diff
@@ -167,7 +167,7 @@ public void setUnprocessedGetInputToNullValue(DeploymentTopology deploymentTopol
      * @return A unified map of input for the topology containing the inputs from the deployment setup as well as the ones coming from location or application
      *         meta-properties.
      */
-    private Map<String, PropertyValue> getInputs(DeploymentTopology deploymentTopology, ApplicationEnvironment environment) {
+    public Map<String, PropertyValue> computeInputs(DeploymentTopology deploymentTopology, ApplicationEnvironment environment) {
```
Comment (lucboutier) [inline alien4cloud-core/src/main/java/alien4cloud/deployment/InputsPreProcessorService.java]: Not sure that this is an helping change.. but doesn't cost neither...

File: alien4cloud-core/src/main/java/alien4cloud/deployment/DeploymentTopologyService.java [hunk 4]
```diff
@@ -562,11 +580,9 @@ private void checkGroups(Map<String, String> groupsLocationsMapping) {
      * @return
      */
     public Map<String, PropertyValue> processForDeployment(DeploymentTopology deploymentTopology, ApplicationEnvironment environment) {
```
Comment (lucboutier) [inline alien4cloud-core/src/main/java/alien4cloud/deployment/DeploymentTopologyService.java]: This sounds like a huge change and I'm quite afraid of the impacts here.

Previous code was actually changing the deployment topology through injectInputValues meaning that the deployment topology had no more get_input on properties but only values. This method is used just before deployment meaning that the orchestrator plugin didn't had to manage inputs.

The current code does not do that anymore. I am really afraid of consequences here.

Comment (igorng) [inline alien4cloud-core/src/main/java/alien4cloud/deployment/DeploymentTopologyService.java]: > Previous code was actually changing the deployment topology through injectInputValues meaning that the deployment topology had no more get_input on properties but only values

This is done now when setting the location or updating an input value.
So when calling this method, the inputs that have values have already been injected in the deployment topology

**Pull request merged**